opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1362,73 +1362,117 @@ template cpgrid::Entity<3> Dune::createEntity(const CpGrid&, int, bool);
-    l1_child_to_parent_cells.resize(data_[1]-> size(0));
-    for (int cell = 0; cell < data_[1]->size(0); ++cell){
-        l1_child_to_parent_cells[cell] = no_parent;
+    const std::vector<int> no_children = {-1};    
+    const std::tuple<int,std::vector<int>> has_no_children = std::make_tuple(-1, no_children);
+    for (int cell = 0; cell < data_[0]-> size(0); ++cell){
+        l0_parent_to_children_cells[cell] = has_no_children;
     }
-    // Rewrite entries for actual parents in level0 and actual children in level1.
-    for (const auto& [parent, children] : parent_to_children_cells) {
-        l0_parent_to_children_cells[parent] = {1, children}; // {level LGR, {child0, child1, ...}}
-        for (const auto& child : children){
-            l1_child_to_parent_cells[child] = {0, parent}; // {level of parent cell, parent cell index in that level}
+    // Get patches corner, face, and cell indices.
+    for (int patch = 0; patch < num_patches; ++patch){
+        const auto& [corners, faces, cells] = (*(this->data_[0])).getPatchGeomIndices(startIJK_vec[patch], endIJK_vec[patch]);
+        all_patch_corners.reserve(corners.size());
+        for (const auto& corner : corners){
+            all_patch_corners.push_back(corner);
         }
-    }
-    // LEVEL 1, definition/declaration of some members:
-    (*data_[1]).level_data_ptr_ = &(this -> data_);
-    (*data_[1]).level_ = 1;
+        all_patch_faces.reserve(faces.size());
+        for (const auto& face : faces){
+            all_patch_faces.push_back(face);
+        }
+        all_patch_cells.reserve(cells.size());
+        for (const auto& cell : cells){
+            all_patch_cells.push_back(cell);
+        }
+        // Build each LGR from the selected patche of cells from level0 (level0 = this->data_[0]).
+        const auto& [level_ptr, boundary_old_to_new_corners, boundary_old_to_new_faces, parent_to_children_faces,
+                     parent_to_children_cells, child_to_parent_faces, child_to_parent_cells]
+            = (*(this-> data_[0])).refinePatch(cells_per_dim_vec[patch], startIJK_vec[patch], endIJK_vec[patch]);
+        // Add each LGR to "data".
+        (this-> data_).push_back(level_ptr);
+        (*data_[patch +1]).level_data_ptr_ = &(this -> data_); // shifted +1 since level0 is coarse grid. Levels are 1,2,..., num_patches.
+        (*data_[patch +1]).level_ = patch +1;
+        //
+        // POPULATING (*data_[0]).parent_to_children_cells_
+        // POPULATING (*data_[patch +1]).child_to_parent_cells_
+        // Extension of container all_patch_child_to_parent_cells, adding {-1,-1} entries for each NO-CHILD cell, in each level/patch.
+        (all_patch_child_to_parent_cells[patch]).resize(child_to_parent_cells.size()); // To consider all cells in the LGR.
+        // True parent cells have entries: {level of the LGR the parent cell has its children, {child0, child1, ...}}
+        // False parent cells (NO PARENT CELLS) have {-1,{-1}} entries.
```

Comment:
Ah, I see. Similar to above my question is: Would `{-1, {}}` also work? Seems cleaner.

Reply:
Yes, {-1,{}} would work for "no-parent-cell" entries in parent_to_children_cells_